src/test_l1_hdmi_cec_driver.c

```diff
@@ -1482,71 +1473,66 @@ void test_hdmicec_hal_l1_hdmiCecTx_sinkDevice_negative( void )
+    if (HDMI_CEC_IO_SENT_FAILED != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
 
     //Set logical address for TV
     //logicalAddress = 0;
     result = HdmiCecAddLogicalAddress(handle, logicalAddress);
-    if (HDMI_CEC_IO_SUCCESS != result) {
-        /*Cleanup before exiting */
-        result = HdmiCecClose (handle);
-        if (HDMI_CEC_IO_SUCCESS != result) { UT_FAIL ("HdmiCecAddLogicalAddress failed"); }
-        UT_FAIL ("HdmiCecAddLogicalAddress failed");
-    }
+    if (HDMI_CEC_IO_SUCCESS != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecAddLogicalAddress failed"); }
 
     buf[0] = 0x0F; UT_LOG ("\n hdmicec buf: 0x%x\n", buf[0]);
 
     //Get logical address
     result = HdmiCecGetLogicalAddress(handle, &logicalAddress);
-    if (HDMI_CEC_IO_SUCCESS != result) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecGetLogicalAddress failed"); }
+    if (HDMI_CEC_IO_SUCCESS != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecGetLogicalAddress failed"); }
 
     /* Invalid input */
     result = HdmiCecTx(handle, buf, len, NULL);
-    if (HDMI_CEC_IO_INVALID_ARGUMENT != result) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecTx failed"); }
+    if (HDMI_CEC_IO_INVALID_ARGUMENT != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
 
     /* Invalid input */
     result = HdmiCecTx(handle, NULL, len, &ret);
-    if (HDMI_CEC_IO_INVALID_ARGUMENT != result) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecTx failed"); }
+    if (HDMI_CEC_IO_INVALID_ARGUMENT != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
 
     /* Invalid input */
     result = HdmiCecTx(0, buf, len, &ret);
-    if (HDMI_CEC_IO_INVALID_HANDLE != result) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecTx failed"); }
+    if (HDMI_CEC_IO_INVALID_HANDLE != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
 
     /* Invalid input */
     result = HdmiCecTx(handle, buf, INT_MIN, &ret);
-    if (HDMI_CEC_IO_INVALID_ARGUMENT != result) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecTx failed"); }
+    if (HDMI_CEC_IO_INVALID_ARGUMENT != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
 
     UT_LOG ("\n hdmicec logicalAddress: 0x%x\n", (logicalAddress&0xFF)<<4);
     buf[0] = ((logicalAddress&0xFF)<<4)|0x0F; UT_LOG ("\n hdmicec buf: 0x%x\n", buf[0]);
 
     /* Positive result */
     result = HdmiCecTx(handle, buf, len, &ret);
-    if (HDMI_CEC_IO_SUCCESS != result) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecTx failed"); }
-    if (HDMI_CEC_IO_SENT_AND_ACKD != ret) { HDMICEC_ASSERT_AUTO_TERM(); UT_FAIL ("HdmiCecTx failed"); }
+    if (HDMI_CEC_IO_SUCCESS != result) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
+    if (HDMI_CEC_IO_SENT_AND_ACKD != ret) { HDMICEC_TERM(); UT_FAIL_FATAL ("HdmiCecTx failed"); }
 
     buf [1] = CEC_DEVICE_VENDOR_ID;
     /*Back to back send and ensure send is not failed.*/
```

Comment:
Its negative test, step 10 and 11 not required. This can be added in positive test case (back to back is not there in positive test case)